packages/ember-simple-auth/addon/services/session.js

```diff
@@ -368,5 +368,29 @@ export default Service.extend(Evented, {
   */
   handleInvalidation(routeAfterInvalidation) {
     handleSessionInvalidated(getOwner(this), routeAfterInvalidation);
-  }
+  },
+
+  /**
+    Setups session service.
+
+    This method must be called for ESA to work properly,
+    previously it was done automatically inside initilizers.
+
+    @method setup
+    @param {callback} callback to run after session restoration,
+    usually the contents of beforeModel should go here.
+    @public
+  */
+  setup(callback) {
```

Comment:
It feels like this method doesn't need to receive a callback, but something like this should do instead:
```js
async beforeModel(transition) {
  await this.session.restore();
  ...
}
```
But I'm not sure if this is something we'd want as it's expected for `restore` to reject in case an authenticator does so, so it'd mean that users would have to actually wrap the `await this.session.restore()` inside `try catch` which might not make sense.